File: projectforge-caldav/src/main/java/org/projectforge/caldav/model/Meeting.java
package org.projectforge.caldav.model;

import java.util.Date;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import io.milton.annotations.ModifiedDate;
import io.milton.annotations.Name;
import io.milton.annotations.UniqueId;

public class Meeting
{
  private static Logger log = LoggerFactory.getLogger(Meeting.class);

  private String uid;
  private Calendar calendar;
  private String name; // filename for the meeting. Must be unique within the user
  private Date createDate;
  private Date modifiedDate;
  private byte[] icalData;

  public Meeting(Calendar calendar)
  {
    this.calendar = calendar;
  }

  @UniqueId
  public String getUniqueId()
  {
    return uid;
  }

  @ModifiedDate
  public Date getModifiedDate()
  {
    return modifiedDate;
  }

  public Calendar getCalendar()
  {
    return calendar;
  }

  public void setCalendar(Calendar calendar)
  {
    this.calendar = calendar;
  }

  @Name
  public String getName()
  {
    return name;
  }

  public void setName(String name)
  {
    this.name = name;
  }

  public String getUid()
  {
    return uid;
  }

  public void setUid(String uid)
  {
    this.uid = uid;
  }

  public void setModifiedDate(Date modifiedDate)
  {
    this.modifiedDate = modifiedDate;
  }

  public byte[] getIcalData()
  {
    return icalData;
  }

  public void setIcalData(byte[] icalData)
  {
    this.icalData = icalData;
  }

  public void setCreateDate(Date createDate)
  {
    this.createDate = createDate;
  }

  public Date getCreateDate()
  {
    return createDate;
  }
}

File: projectforge-caldav/src/main/kotlin/org/projectforge/caldav/model/Meeting.kt
/////////////////////////////////////////////////////////////////////////////
//
// Project ProjectForge Community Edition
//         www.projectforge.org
//
// Copyright (C) 2001-2024 Micromata GmbH, Germany (www.micromata.com)
//
// ProjectForge is dual-licensed.
//
// This community edition is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License as published
// by the Free Software Foundation; version 3 of the License.
//
// This community edition is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
// Public License for more details.
//
// You should have received a copy of the GNU General Public License along
// with this program; if not, see http://www.gnu.org/licenses/.
//
/////////////////////////////////////////////////////////////////////////////

package org.projectforge.caldav.model

import io.milton.annotations.ModifiedDate
import io.milton.annotations.Name
import io.milton.annotations.UniqueId
import org.slf4j.LoggerFactory
import java.util.*

class Meeting(var calendar: Calendar) {
    var uniqueId: String? = null
        @UniqueId get() = field
        set

    // filename for the meeting. Must be unique within the user
    @get:Name
    var name: String? = null
    var createDate: Date? = null
    @get:ModifiedDate
    var modifiedDate: Date? = null
    var icalData: ByteArray? = null

    companion object {
        private val log = LoggerFactory.getLogger(Meeting::class.java)
    }

}
